File: linux/ardium_source/presentation/10_Advanced/advanced.m
// Topic: Advanced Global Scope
// Manage Global State

#import <Foundation/Foundation.h>

static NSString *AppStatus = @"Active";
static int AppUsers = 0;

int main(int argc, const char * argv[]) {
    @autoreleasepool {
        NSLog(@"Current Status: %@", AppStatus);
        
        AppStatus = @"Maintenance";
        
        NSLog(@"New Status: %@", AppStatus);
    }
    return 0;
}

File: linux/ardium_source/presentation/10_Advanced/advanced.go
// Topic: Advanced Global Scope
// Manage Global State

package main

import "fmt"

var AppState = struct {
	Status string
	Users  int
}{
	Status: "Active",
	Users:  0,
}

func main() {
	fmt.Printf("Current Status: %s\n", AppState.Status)

	AppState.Status = "Maintenance"

	fmt.Printf("New Status: %s\n", AppState.Status)
}
